lib/ansible/galaxy/collection/__init__.py

```diff
@@ -674,60 +674,10 @@ def install_collections(
     :param force_deps: Re-install a collection as well as its dependencies if they have already been installed.
     """
     existing_collections = {
-        Requirement(coll.fqcn, coll.ver, coll.src, coll.type, None)
-        for coll in find_existing_collections(output_path, artifacts_manager)
-    }
-
-    unsatisfied_requirements = set(
-        chain.from_iterable(
-            (
-                Requirement.from_dir_path(sub_coll, artifacts_manager)
-                for sub_coll in (
-                    artifacts_manager.
-                    get_direct_collection_dependencies(install_req).
-                    keys()
-                )
-            )
-            if install_req.is_subdirs else (install_req, )
-            for install_req in collections
-        ),
-    )
-    requested_requirements_names = {req.fqcn for req in unsatisfied_requirements}
-
-    # NOTE: Don't attempt to reevaluate already installed deps
-    # NOTE: unless `--force` or `--force-with-deps` is passed
-    unsatisfied_requirements -= set() if force or force_deps else {
-        req
-        for req in unsatisfied_requirements
-        for exs in existing_collections
-        if req.fqcn == exs.fqcn and meets_requirements(exs.ver, req.ver)
-    }
-
-    if not unsatisfied_requirements and not upgrade:
-        display.display(
-            'Nothing to do. All requested collections are already '
-            'installed. If you want to reinstall them, '
-            'consider using `--force`.'
-        )
-        return
-
-    # FIXME: This probably needs to be improved to
-    # FIXME: properly match differing src/type.
-    existing_non_requested_collections = {
-        coll for coll in existing_collections
-        if coll.fqcn not in requested_requirements_names
-    }
-
-    preferred_requirements = (
-        [] if force_deps
-        else existing_non_requested_collections if force
-        else existing_collections
-    )
-    preferred_collections = {
-        # NOTE: No need to include signatures if the collection is already installed
         Candidate(coll.fqcn, coll.ver, coll.src, coll.type, None)
-        for coll in preferred_requirements
+        for coll in find_existing_collections(output_path, artifacts_manager)
```

Comment:
We shouldn't start scanning the disk `if force_deps` — all this is being discarded on the next line in 50% of cases.